src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -281,17 +282,23 @@ private synchronized void doConnect() throws MqttException, KeyStoreException {
     private void reconnectAndResubscribe() {
         int waitBeforeRetry = MIN_WAIT_RETRY_IN_SECONDS;
 
-        while (!mqttClientInternal.isConnected()) {
+        while (!mqttClientInternal.isConnected() && !Thread.currentThread().isInterrupted()) {
             try {
                 // TODO: Clean up this loop
                 doConnect();
             } catch (MqttException | KeyStoreException e) {
+                if (Utils.getUltimateCause(e) instanceof InterruptedException) {
+                    // paho doesn't reset the interrupt flag
+                    Thread.currentThread().interrupt();
```

Comment:
logging is appropriate, just don't log the cause because that will dump a big and useless stacktrace.

Reply:
gotcha, adjusted